assets/src/edit-story/components/canvas/karma/multiSelectionMovable.karma.js

```diff
@@ -0,0 +1,171 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { Fixture } from '../../../karma';
+import { useStory } from '../../../app/story';
+import useInsertElement from '../useInsertElement';
+import { TEXT_ELEMENT_DEFAULT_FONT } from '../../../app/font/defaultFonts';
+
+describe('Multi-selection Movable integration', () => {
+  let fixture;
+
+  beforeEach(async () => {
+    fixture = new Fixture();
+
+    await fixture.render();
+  });
+
+  afterEach(() => {
+    fixture.restore();
+  });
+
+  describe('multi-selection', () => {
+    let element1;
+    let element2;
+    let element3;
+    let frame1;
+    let frame2;
+    let frame3;
+
+    beforeEach(async () => {
+      const insertElement = await fixture.renderHook(() => useInsertElement());
+      element1 = await fixture.act(() =>
+        insertElement('text', {
+          font: TEXT_ELEMENT_DEFAULT_FONT,
+          content: 'Text A',
+          x: 10,
+          y: 10,
+          width: 50,
+          height: 50,
+        })
+      );
+
+      element2 = await fixture.act(() =>
+        insertElement('text', {
+          font: TEXT_ELEMENT_DEFAULT_FONT,
+          content: 'Text B',
+          x: 100,
+          y: 100,
+          width: 50,
+          height: 50,
+        })
+      );
+
+      element3 = await fixture.act(() =>
+        insertElement('text', {
+          font: TEXT_ELEMENT_DEFAULT_FONT,
+          content: 'Text B',
+          x: element1.x + element1.width + 1,
+          y: element1.y + element1.height + 1,
+          width: 20,
+          height: 20,
+        })
+      );
+      frame3 = fixture.querySelector(
+        `[data-element-id="${element3.id}"] [data-testid="textFrame"]`
+      );
+
+      frame1 = fixture.querySelector(
+        `[data-element-id="${element1.id}"] [data-testid="textFrame"]`
+      );
+
+      frame2 = fixture.querySelector(
+        `[data-element-id="${element2.id}"] [data-testid="textFrame"]`
+      );
+    });
+
+    it('should render initial content', () => {
+      expect(frame1.textContent).toEqual('Text A');
+      expect(frame2.textContent).toEqual('Text B');
+    });
+
+    it('should multi-select elements on Shift + click', async () => {
+      await fixture.events.click(frame1);
+      await fixture.events.keyboard.down('Shift');
+      await fixture.events.click(frame2);
+      await fixture.events.keyboard.up('Shift');
+
+      const storyContext = await fixture.renderHook(() => useStory());
+      expect(storyContext.state.selectedElementIds).toEqual([
+        element1.id,
+        element2.id,
+      ]);
+    });
+
+    describe('click interaction', () => {
+      let safezone;
+      beforeEach(async () => {
+        await fixture.events.click(frame1);
```

Comment:
This is the same code as above in the `it('should multi-select elements on Shift + click')`. We can do like this:

```
describe('multi-select', () => {
  beforeEach(() => {
    // multi-selection events
  });
  it('should start with multi-selection', () => {
    expect(...).
  });
  it('should select on element...', () => {});
  it('should select bg element...', () => {});
})
```